src/librustdoc/doctest/make.rs

```diff
@@ -299,15 +317,32 @@ fn parse_source(
     let dcx = DiagCtxt::new(Box::new(emitter)).disable_warnings();
     let psess = ParseSess::with_dcx(dcx, sm);
 
-    let mut parser = match new_parser_from_source_str(&psess, filename, source) {
+    let mut parser = match new_parser_from_source_str(&psess, filename, wrapped_source) {
         Ok(p) => p,
         Err(errs) => {
             errs.into_iter().for_each(|err| err.cancel());
             cancel_error_count(&psess);
-            return ParsingResult::Failed;
+            return Err(());
         }
     };
-    let mut parsing_result = ParsingResult::Ok;
+
+    fn push_to_s(
+        s: &mut String,
+        source: &str,
+        span: rustc_span::Span,
+        prev_span_hi: &mut Option<usize>,
```

Comment:
```suggestion
        prev_span_hi: &mut usize,
```

this could just be an `usize` right? starting off at `0` instead of `None` in `parse_source`

Reply:
Fair enough! I do `.unwrap_or(0)` with it so it's definitely useless to keep it in an `Option`...